libs/@local/codegen/tests/codegen/snapshots/codegen__AdjacentlyTaggedEnum::Typescript.snap

```diff
@@ -0,0 +1,30 @@
+---
+source: libs/@local/codegen/tests/codegen/main.rs
+expression: generated
+---
+type AdjacentlyTaggedEnum = {
+	type: "Unit"
+} | {
+	type: "EmptyUnnamed"
+	content: []
+} | {
+	type: "SingleUnnamed"
+	content: number
+} | {
+	type: "DoubleUnnamed"
+	content: [boolean, string]
+} | {
+	type: "EmptyNamed"
+	content: {}
```

Comment:
This might be incorrect actually, this might need to be `Record<string, never>` instead, `{}` admits *any* record in TypeScript. See: https://github.com/microsoft/TypeScript/wiki/FAQ#--does-not-refer-to-objects-with-no-properties

Reply:
It's not incorrect, just not ideal. `{}` is a valid type for an empty map. I agree that we could add an edge case here, but I'd prefer to do this in a follow up: H-4487. It's not an important thing because it does not happen regularly. But to use it, the type _must_ be specified with `deny_unknown_fields`, otherwise, `Record<string, never>` would be wrong